library-app/src/components/Books/BooksList/ManageBookForm.tsx

```diff
@@ -0,0 +1,159 @@
+import axios from 'axios'
+import { FormEvent, useEffect, useState } from 'react'
+import { postBookRequest } from '../../../services/BooksServices'
+import styles from './ManageBookForm.module.css'
+import Select, { MultiValue } from 'react-select'
+import BookBodyData from '../../../models/bookData.model'
+import Author from '../../../models/author.model'
+import getAuthors from '../../../services/AuthorServices'
+import placeholder from '../../../assets/placeholderImg/placeholder.jpeg'
+
+const ManageBookForm = () => {
+  const [authors, setAuthors] = useState<Author[]>([])
+  const [requestCover, setRequestCover] = useState<Blob>(new Blob())
+  const [cover, setCover] = useState('')
+  const [formData, setFormData] = useState<BookBodyData>({
+    Id: 0,
+    Title: '',
+    Description: '',
+    Isbn: '',
+    Quantity: 0,
+    Cover: requestCover,
+    PublishDate: '',
+    AuthorIds: [],
+  })
+
+  const handleFileChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
+        reader.readAsDataURL(files[0])
+        setRequestCover(files[0])
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data) setCover(base64data as string)
+        }
+      }
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await getAuthors().then((response) => {
```

Comment:
Why both await and then?

Reply:
Also, you should implement fetchData outside of useEffect and here you will only call it

Reply:
I thought it's the right way. Now i understand that then, without async/await , is also asynchronous function